### ISSUE TITLE
Invalid imports created when generating code for an object using more than one goType

### ISSUE THREAD

**Issue by rltvty** [ent/ent#2685]
- [x] The issue is present in the latest release.
- [x] I have searched the [issues](https://github.com/ent/ent/issues) of this repository and believe that this is not a duplicate.

might be related to https://github.com/ent/ent/issues/1565

## Current Behavior 😯

In the attached bug-repo, there are two versioned proto packages : 
* `ent_multiple_proto_bug.bar.v1alpha`
* `ent_multiple_proto_bug.foo.v1alpha`

The generated code for these libraries end up at the following paths:
* go package: `barv1alpha` path: `gen/ent-multiple-proto-bug/bar/v1alpha`
* go package: `foov1alpha` path: `gen/ent-multiple-proto-bug/foo/v1alpha`

The *ValueScanner* interface is implemented for both packages in the `*.pb.ent.go` files. 

Both packages are used in the `ent/schema/user.go` object:
```
func (User) Fields() []ent.Field {
	return []ent.Field{
		field.Int("age"),
		field.String("name"),
		field.Bytes("foo").GoType(&foov1alpha.Foo{}),
		field.Bytes("bar").GoType(&barv1alpha.Bar{}),
	}
}
```
(see: https://github.com/rltvty/ent-multiple-proto-bug/blob/master/ent/schema/user.go for full code)

After running `go generate ./ent`,  the generated code has invalid imports and will not compile. In all cases, an extra, non-aliased, import of `"github.com/rltvty/ent-multiple-proto-bug/gen/ent-multiple-proto-bug/foo/v1alpha"` seems to be the issue.

The following files in the `./ent` folder have the issue:
* `mutation.go`: https://github.com/rltvty/ent-multiple-proto-bug/blob/master/ent/mutation.go#L14
* `runtime.go`: https://github.com/rltvty/ent-multiple-proto-bug/blob/master/ent/runtime.go#L6
* `user_delete.go`: https://github.com/rltvty/ent-multiple-proto-bug/blob/master/ent/user_delete.go#L14
* `user_query.go`: https://github.com/rltvty/ent-multiple-proto-bug/blob/master/ent/user_query.go#L15
* `user_update.go`: https://github.com/rltvty/ent-multiple-proto-bug/blob/master/ent/user_update.go#L16
* `user.go`: https://github.com/rltvty/ent-multiple-proto-bug/blob/master/ent/user.go#L12

## Expected Behavior 🤔

Expected that the generated code doesn't have this extra import statement, and the code can compile.

if I delete these extra imports, I have confirmed that the code does compile and work as expected 

## Steps to Reproduce 🕹

https://github.com/rltvty/ent-multiple-proto-bug

| Tech        | Version |
| ----------- | ------- |
| Go          | 1.18.3  |
| Ent         | v0.10.2   |

**Comment by a8m** [comment]
Added https://github.com/ent/ent/pull/2686 which should address this issue. I tested it on your repository and it worked fine, would you mind checking it too? You can `go get` a specific branch or commit to try this. 

I'll add some tests for the PR tomorrow and will merge it 😎

**Comment by a8m** [comment]
Closed with #2686. Please, update to the latest master using `go get entgo.io/ent@master` and let me know if you have any troubles with it. Thanks
